File: application/lib/types/inquiry.ts
export type InquiryCategory = "BUG" | "SUGGESTION" | "QUESTION";

export interface InquiryResponse {
  id: number;
  userId: number;
  nickname: string;
  category: InquiryCategory;
  title: string;
  content: string;
  answer: string | null;
  answeredAt: string | null;
  createdAt: string;
}

export interface InquiryCreateRequest {
  category: InquiryCategory;
  title: string;
  content: string;
}

export const INQUIRY_CATEGORY_LABEL: Record<InquiryCategory, string> = {
  BUG: "오류 신고",
  SUGGESTION: "기능 제안",
  QUESTION: "이용 문의",
};

File: backend/src/main/java/com/back/domain/inquiry/entity/Inquiry.java
package com.back.domain.inquiry.entity;

import com.back.domain.user.user.entity.User;
import jakarta.persistence.*;
import java.time.LocalDateTime;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
@Table(name = "inquiries")
@EntityListeners(AuditingEntityListener.class)
public class Inquiry {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private InquiryCategory category;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    @Column(columnDefinition = "TEXT")
    private String answer;

    private LocalDateTime answeredAt;

    @CreatedDate
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    public void answer(String answer) {
        this.answer = answer;
        this.answeredAt = LocalDateTime.now();
    }
}
